### PR TITLE
Md 4 video on page dom

### PR DESCRIPTION
**Pull request by ThatOneGoat**
Button appnds video element

### CODE REVIEW

File: js/content.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+var counter = 0;
+var btn = document.createElement("BUTTON");
+var txt = document.createTextNode("michael_click");
+var image = document.createElement("img");
+// var video = document.createElement("video");
+
+// video.autoplay = true;
```
Comment (mmattikalli) [inline js/content.js]: I think you might need to delete all these comments - seem unnecessary

File: js/content.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
+var counter = 0;
+var btn = document.createElement("BUTTON");
+var txt = document.createTextNode("michael_click");
```
Comment (mmattikalli) [inline js/content.js]: Please change the name of your button from "Michael_click"

File: js/faceID.js [hunk 3]
```diff
@@ -90,44 +89,38 @@ function captureFrame(video) {
...
+                        });
+                    }
                 });
-            });
         }, 8000);
     }
 });
 
-
-document.getElementById('enableFaceIdScreen').addEventListener('click', () => {
```
Comment (mmattikalli) [inline js/faceID.js]: add more comments to clarify what it is doing

File: manifest.json [hunk 4]
```diff
@@ -17,15 +19,22 @@
     "default_title": "WebAssist",
     "default_popup": "html/popup.html"
   },
-  "content_scripts": [{
-    "matches": [
+  "content_scripts": [
+    {
+      "matches": [
         "<all_urls>"
+      ],
```
Comment (mmattikalli) [inline manifest.json]: I thinkt he formatting might be messed up a little - make sure to format

File: js/content.js [hunk 5]
```diff
@@ -1,7 +1,118 @@
...
+            video.style.width = "320px";
+            video.style.height = "240px";
+            video.style.position = "fixed";
+            video.style.bottom = "10px";
+            video.style.left = "10px";
+            video.style.borderRadius = "5px";
+            video.style.zIndex = "10000000000";
+            document.body.appendChild(video);
+
+            console.log("hola");
```
Comment (Kaboomboom3) [inline js/content.js]: Remove debug print

File: js/content.js [hunk 6]
```diff
@@ -1,7 +1,118 @@
...
+
+            console.log(navigator.mediaDevices.getUserMedia);
+            if (navigator.mediaDevices.getUserMedia) {
+                //dfghjk
+                console.log(navigator.platform);
+                //console.log(navigator.mediaDevices);
+                navigator.mediaDevices.getUserMedia({
+                    video: true
+                }).then(function (stream) {
+                    console.log("made it through");
```
Comment (Kaboomboom3) [inline js/content.js]: Remove debug print

File: js/content.js [hunk 7]
```diff
@@ -1,7 +1,118 @@
...
+            video.style.left = "10px";
+            video.style.borderRadius = "5px";
+            video.style.zIndex = "10000000000";
+            document.body.appendChild(video);
+
+            console.log("hola");
+
+            console.log(navigator.mediaDevices.getUserMedia);
+            if (navigator.mediaDevices.getUserMedia) {
+                //dfghjk
```
Comment (Kaboomboom3) [inline js/content.js]: ???

File: js/content.js [hunk 8]
```diff
@@ -1,7 +1,118 @@
...
+                    console.log("made it through");
+                    var video = document.getElementById('vid');
+                    video.srcObject = stream;
+                    // video.onloadedmetadata = function (e) {
+                    //     video.play();
+                    // }
+                }).catch(function (e) { console.log(e); })
+            }
+            if (request.greeting == "hello")
+                sendResponse({ farewell: "goodbye" });
```
Comment (Kaboomboom3) [inline js/content.js]: This isn't clear, what does this mean?

File: js/content.js-michael [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+btn.style.position = "fixed";
+btn.style.bottom = "10px";
+btn.style.left = "10px";
+
+btn.appendChild(txt);
+// document.body.appendChild(image);
+document.body.appendChild(btn);
+btn.onclick = function () {
+    console.log("hola");
```
Comment (Kaboomboom3) [inline js/content.js-michael]: Remove debug print

File: js/faceID.js [hunk 10]
```diff
@@ -90,44 +89,38 @@ function captureFrame(video) {
...
-                    faceJS.verifyFace(calibratedId, face.faceId)
-                    .then(result => {
-                        console.log("itworked");
-                    });
+                .then(faces => {
+                    if (faces.length < 1) {
+                        alert("ur face gone");
+                    } else {
+                        faces.forEach(face => {
+                            console.log("itworkedtoreachforloop");
```
Comment (Kaboomboom3) [inline js/faceID.js]: Remove debug print

File: js/faceID.js [hunk 11]
```diff
@@ -90,44 +89,38 @@ function captureFrame(video) {
...
-    if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
-        navigator.mediaDevices.getUserMedia({
-            video: true
-        }).then(function (stream) {
-            video.src = window.URL.createObjectURL(stream);
-            video.play();
+document.getElementById("enableFaceSecurity").addEventListener("click", () => {
+    browser.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+        browser.tabs.sendMessage(tabs[0].id, { greeting: "hello" }, function (response) {
+            console.log(response.farewell);
```
Comment (Kaboomboom3) [inline js/faceID.js]: Remove debug print

File: js/content.js [hunk 12]
```diff
@@ -31,8 +31,8 @@ browser.runtime.onMessage.addListener(
                     video.srcObject = stream;
                 }).catch(function (e) { console.log(e); }) //error catch
             }
-            if (request.greeting == "hello") //Send a response back to extension script
-                sendResponse({ farewell: "goodbye" });
+            if (request.type == "Get Video") //Send a response back to extension script
```
Comment (Kaboomboom3) [inline js/content.js]: Use GetVideo to match messages sent to background script

File: js/faceID.js [hunk 13]
```diff
@@ -121,8 +121,8 @@ document.querySelector('#calibrate').addEventListener('click', () => {
 document.getElementById("enableFaceSecurity").addEventListener("click", () => {
     browser.tabs.query({ active: true, currentWindow: true }, function (tabs) { //Get tabs with specified properties
         //Send a message out to get a response
-        browser.tabs.sendMessage(tabs[0].id, { greeting: "hello" }, function (response) {
-            console.log(response.farewell);
+        browser.tabs.sendMessage(tabs[0].id, { type: "Get Video" }, function (response) {
```
Comment (Kaboomboom3) [inline js/faceID.js]: Use GetVideo to match messages sent to background script

File: manifest.json [hunk 14]
```diff
@@ -1,14 +1,14 @@
-
 {
-  "manifest_version": 2,
   "name": "WebAssist",
   "author": "Microsoft Edge Extension Developer",
   "version": "1.1",
+  "manifest_version": 2,
   "description": "Assistance for your web browser",
   "permissions": [
-    "<all_urls>",
```
Comment (mmattikalli) [inline manifest.json]: We need the <all_urls> tag

File: manifest.json [hunk 15]
```diff
@@ -1,14 +1,14 @@
...
-    "activeTab",
-    "tabs"
+    "*",
```
Comment (mmattikalli) [inline manifest.json]: This needs to be taken out - we want all URL's to be accessed, but not all permissions

File: js/faceID.js [hunk 16]
```diff
@@ -90,44 +89,40 @@ function captureFrame(video) {
...
-            calibratedId = faces[0].faceId;
-        });
-
-        // capture("calibrating");
+            .then(faces => {
+                calibratedId = faces[0].faceId;
+            });
         counter = 1;
         setInterval(() => {
-            // console.log("ticking");
```
Comment (mmattikalli) [inline js/faceID.js]: remove console.logs

File: manifest.json [hunk 17]
```diff
@@ -1,14 +1,14 @@
-
 {
-  "manifest_version": 2,
   "name": "WebAssist",
   "author": "Microsoft Edge Extension Developer",
   "version": "1.1",
+  "manifest_version": 2,
```
Comment (mmattikalli) [inline manifest.json]: Put in comment that says this manifest version is for chrome

File: js/faceID.js (hunk 16 above)
Comment (mmattikalli) [reply js/faceID.js]: sorry

**Pull request merged**